src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1143,6 +1176,9 @@ RuntimeFieldHandleInternal Value
     [StructLayout(LayoutKind.Sequential)]
     internal sealed class RuntimeFieldInfoStub : IRuntimeFieldInfo
     {
+        public RuntimeFieldInfoStub(IntPtr ptr)
+            => m_fieldHandle = new RuntimeFieldHandleInternal(ptr);
+
         // These unused variables are used to ensure that this class has the same layout as RuntimeFieldInfo
 #pragma warning disable 414, 169
         private object? m_keepalive;
```

Comment:
Does this need to initialize KeepAlive, same as `RuntimeMethodHandle` ?

Reply:
That is a good question. This wasn't done previously so I don't think so, but it is possible this has a bug in its use. I'm not entirely sure the reason for the keep alive logic so if you can share or perhaps point me to where an example of need is that would be helpful.

Reply:
The keep alive logic is for collectible assemblies. It is there to ensure that the RuntimeFieldHandle will prevent the assembly from getting unloaded.

Reply:
> This wasn't done previously so I don't think so

The only is place where I see RuntimeFieldInfoStub created is here: https://github.com/dotnet/runtime/blob/6d21c938b2d420b9e06107e409aec7a01b376f59/src/coreclr/vm/field.cpp#L814-L815 . It is setting keeplive correctly.